src/auth.rs

```diff
@@ -539,6 +525,24 @@ pub async fn verify_identity(
     })
 }
 
+fn parse_cacao_statement(statement: &str, domain: &str) -> Result<AuthorizedApp> {
```

Comment:
should be in a different PR as it seems is not related to the rate limiting

Reply:
It's related because the rate limit prevents more than 5 `/notify` calls in a single second, and the tests all use the same project ID. So had to refactor the type of tests to allow them to pass.